frappe/public/js/frappe/upload.js

```diff
@@ -109,7 +109,7 @@ frappe.upload = {
 				}, function(values) {
 					args["is_private"] = values.is_private;
 					frappe.upload.upload_to_server(fileobj, args, opts, dataurl);
-				}, __("Private or Public?"));
+				}, __("Make ") + fileobj.name + __(" private or public?"));
 			} else {
```

Comment:
Will this add multiple popups? Why not show a grid with all the files and let user "check" the private files?